server/models/questions/getQuestion.js

```diff
@@ -0,0 +1,32 @@
+const { pool } = require('../database')
+
+/**
+ *
+ * @param {*} param0
+ * @param {String} param0.username
+ * @param {Number} param0.questionId
+ * @return {Promise}
+ */
+
+function getQuestion({ username, questionId }) {
+  return new Promise((resolve, reject) => {
+    pool.query(
+      `SELECT * FROM 
+        questions q 
+        LEFT JOIN questions_editors qe ON q.id = qe.question_id
+        WHERE q.id=? AND (q.creator = ? OR qe.editor = ?)`,
+      [questionId, username, username],
```

Comment:
Performing a JOIN here can be much expensive and that too, LEFT JOIN. Try to prevent at least LEFT JOIN here, as getQuestion API would be called much frequently, as compared to the CREATE  question. So I would suggest adding an entry for the creator also into the `questions_editors` table, keeping in mind the approx ratio of CREATE question API calls vs. Get a Question API calls. After that, you can use INNER_JOIN here and the query would become :
```
`SELECT * FROM 
        questions q 
        INNER JOIN questions_editors qe ON q.id = qe.question_id
        WHERE q.id=? AND  qe.editor = ?`,
      [questionId, username]
```
Also, this query would be little faster than previous, as it doesn't use the OR operator and OR operator generally performs slower because it doesn't use indexing.

Reply:
I think `JOIN` has to be done..It is much faster than sub-queries...yeah, you are **right** in this regard, I had though as long as there are `non-null `foreign keys, `LEFT` or `INNER JOIN `both would perform equally well. But turns out `INNER JOIN` uses `hashing` which is much faster. I will change this to `INNER JOIN`.

Reply:
Regarding `creator`, u are **right** as well. It might be equally performant in case of `sub-queries` whether `creator` is stored in `questions_editors` or not, but in `JOIN`, `q.creator = ? OR qe.editor = ?` would check for second condition most of the times, and if u think of changing it to `q.editor = ? OR qe.creator = ?`, it would mean that `editor` is being checked every time. So it is **better** to store  `creator` in `questions_editors`